### PR TITLE
bradl3yC - DLC API integration

### PR DESCRIPTION
**Pull request by bradl3yC**
## Description
- [x] - Wire up post endpoint
- [x] - Resolve issues with confirmation page not showing
- [x] - Resolve issue with products showing up in wrong category on review page

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/2346/components/Accessories.jsx [hunk 1]
```diff
@@ -11,26 +11,23 @@ import { ACCESSORIES } from '../constants';
...
     } else {
-      updatedSelectedProducts = selectedProducts.filter(
+      updatedorder = order.filter(
         selectedProduct => selectedProduct.productId !== supply.productId,
       );
     }
     const updatedFormData = {
       ...formData,
-      selectedProducts: updatedSelectedProducts,
+      order: updatedorder,
```
Comment (kelsonic) [inline src/applications/disability-benefits/2346/components/Accessories.jsx:25]: Perhaps make this var name camel case? 🐫

File: src/applications/disability-benefits/2346/components/Batteries.jsx [hunk 2]
```diff
@@ -11,28 +11,25 @@ import { BATTERIES } from '../constants';
 
 class Batteries extends Component {
   handleChecked = (checked, batterySupply) => {
-    const { selectedProducts, formData } = this.props;
-    let updatedSelectedProducts;
+    const { order, formData } = this.props;
+    let updatedorder;
```
Comment (kelsonic) [inline src/applications/disability-benefits/2346/components/Batteries.jsx:15]: Same here 🙂

**Review by maharielrosario — APPROVED** [review]

File: src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx [hunk 3]
```diff
@@ -100,34 +95,50 @@ const ConfirmationPage = ({
...
+  confirmationPageData: PropTypes.shape({
+    fullName: PropTypes.shape({
+      first: PropTypes.string.isRequired,
+      middle: PropTypes.string,
+      last: PropTypes.string.isRequired,
+    }),
+    email: PropTypes.string.isRequired,
+    order: PropTypes.array.isRequired,
+    shippingAddress: PropTypes.object.isRequired,
+    supplies: PropTypes.array.isRequired,
```
Comment (kelsonic) [inline src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx:107]: Missing new prop `vetEmail` I think 🙂

Comment (kelsonic) [reply src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx:107]: Oops, sorry rather we will need to update `email` to `vetEmail` here 👍

File: src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx [hunk 4]
```diff
@@ -100,34 +95,50 @@ const ConfirmationPage = ({
...
   const selectedProductArray = supplies?.filter(supply =>
     productIdArray?.includes(supply.productId),
   );
+
   // Temporary fallback until this is added to the API response
   const submittedAt = state.form?.submission?.submittedAt || moment();
 
   return {
     submittedAt,
-    email: state.form?.data?.email,
```
Comment (kelsonic) [inline src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx:125]: Where is `vetEmail` coming from? Just a passed in prop?

**Review by kelsonic — APPROVED** [review]
LGTM! Lots of good work here! 🎉

File: src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx (hunk 4 above)
Comment (bradl3yC) [reply src/applications/disability-benefits/2346/containers/ConfirmationPage.jsx:125]: Its in state - state.form.data

**Pull request merged**